sdk/ai/azure-ai-inference/azure/ai/inference/prompts/_patch.py

```diff
@@ -0,0 +1,85 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+
+import azure.ai.inference.prompts as prompts
+from .core import Prompty
+from .utils import prepare
+from .mustache import render
+
+
+class PromptTemplate:
+    """The helper class which takes varient of inputs, e.g. Prompty format or string, and returns the parsed prompt in an array.
+
+    :param prompty: Prompty object which contains both model config and prompt template.
+    :type prompty: Prompty
+    :param prompt_template: The prompt template string.
+    :type prompt_template: str
+    :param api: The API type, e.g. "chat" or "completion".
+    :type api: str
+    :param model_name: The model name, e.g. "gpt-4o-mini".
+    :type model_name: str
+    """
+
+    @staticmethod
+    def from_prompty(file_path: str):
+        if not file_path:
+            raise ValueError("Please provide file_path")
+        prompty = prompts.load(file_path)
+        return PromptTemplate(prompty=prompty)        
+    
+    @staticmethod
+    def from_message(
+        prompt_template: str,
+        api: str = "chat",
+        model_name: str | None = None
+    ):
+        return PromptTemplate(api=api, prompt_template=prompt_template, model_name=model_name, prompty=None)
+
+    def __init__(
```

Comment:
Missing `@overloads`?

Reply:
We recommend the users to only use the classmethod `from_prompty` and `from_message`, they shouldn't use the constructor directly unless they check the source code for the constraints below. As Python constructors can't be private, maybe we can leave it this way for now?

Reply:
For a preview, yes. But it would still be good to improve/make it less awkward to call the constructor by making the parameters keyword-only. We should also add a doccomment indicating what the preferred methods for creating an instance is.

Suggestion: 

@typing.overload
def __init__(self, *, prompty: Prompty) -> None: ...

@typing.overload
def __init__(self, *, api: str = 'chat', prompt_template: str, model_name: str) -> None: ...

#(And then the actual implementation)

Reply:
Thanks for all the feedbacks, how about we only keep the 2 `classmethod` but no constructors, what do you think? This is the sample code:

https://github.com/YusakuNo1/azure-sdk-for-python/blob/users/daviwu/prompty-test-overload/sdk/ai/azure-ai-inference/azure/ai/inference/prompts/_patch.py#L21-L44